lib/facter/jenkins.rb

```diff
@@ -12,3 +12,24 @@
     plugins.keys.sort.map { |plugin| "#{plugin} #{plugins[plugin][:plugin_version]}" }.join(', ')
   end
 end
+
+Facter.add(:jenkins_version) do
+  confine kernel: 'Linux'
+
+  setcode do
+    war = case Facter.value('os')['family']
+          when 'Debian'
+            '/usr/share/jenkins'
+          when 'Archlinux'
+            '/usr/share/java/jenkins/'
+          else
+            '/usr/lib/jenkins'
+          end + '/jenkins.war'
```

Comment:
This bit is very well hidden. I think it would be easier to read if you have a `directory = ...` and `war = directory + "/jenkins.war"`.

Reply:
Np :)